### PR TITLE
Summary time period card re-render bug

### PR DESCRIPTION
**Pull request by Kseniia-Koshkina**
Separate the timebank screen cards into their own components and solve the chart flickering problem

### CODE REVIEW

File: src/components/timebank/summary-time-entries-card.tsx [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+              secondary={getHoursAndMinutes(personTotalTime.expected)}
+            />
+          </ListItem>
+        </List>
+      </>
+    );
+  };
+
+  return (
+    <>
```
Comment (Jdallos) [inline src/components/timebank/summary-time-entries-card.tsx:129]: React fragments not needed here?

File: src/utils/chart-utils.tsx [hunk 2]
```diff
@@ -137,25 +137,25 @@ export const renderCustomizedTooltipBarChart = ({
       >
         {name}
       </Typography>
-      {billableProject &&
+      {billableProject !== undefined && billableProject !==0 &&
```
Comment (Jdallos) [inline src/utils/chart-utils.tsx:140]: I don't think these changes to any of the conditions in this file are necessary?

**Review by Jdallos — CHANGES_REQUESTED** [review]
Some minor suggested changes

File: src/utils/chart-utils.tsx (hunk 2 above)
Comment (Kseniia-Koshkina) [reply src/utils/chart-utils.tsx:140]: I think we need these changes. 0 displays in tooltip, when value do not exist (it returned the first part of the falsy expression).

**Review by Jdallos — CHANGES_REQUESTED** [review]
Changes since previous review seem fine, however as it's been a while since that review please test the UI thoroughly. Also there are cloudflare issues that should be resolved before approval

**Review by DZotoff — APPROVED** [review]

File: src/components/timebank/specific-time-entries-card.tsx [hunk 3]
```diff
@@ -102,21 +103,24 @@ const SpecificTimeEntriesCard = ({selectedEmployeeId}: Props) => {
...
-                ? getHoursAndMinutes(
-                    selectedEntries.reduce((prev, next) => prev + Number(next[item.propName]), 0)
-                  )
-                : getHoursAndMinutes(Number(personDailyEntry ? personDailyEntry[item.propName] : 0))
-            }
-          />
-        </ListItem>
-      ))}
+      {timeEntriesListItems.map((item) => {
+        const entryValue = Number(personDailyEntry ? personDailyEntry[item.propName] : 0);
```
Comment (Jdallos) [inline src/components/timebank/specific-time-entries-card.tsx:107]:
```suggestion
        const entryValue = Number(personDailyEntry ? personDailyEntry[item.propName] : 0);
        
```

File: src/components/timebank/specific-time-entries-card.tsx [hunk 4]
```diff
@@ -35,8 +36,8 @@ const SpecificTimeEntriesCard = ({selectedEmployeeId}: Props) => {
   const [byRange, setByRange] = useState({
     dailyEntries: false
   });
-  const [personDailyEntry, setPersonDailyEntry] = useAtom(personDailyEntryAtom);
-  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+  const [personDailyEntry, setPersonDailyEntry] = useState<DailyEntryWithIndexSignature|undefined>(useAtomValue(personDailyEntryAtom));
```
Comment (Jdallos) [inline src/components/timebank/specific-time-entries-card.tsx:39]:
```suggestion
  const [personDailyEntry, setPersonDailyEntry] = useState<DailyEntryWithIndexSignature | undefined>(useAtomValue(personDailyEntryAtom));
```

**Review by Jdallos — APPROVED** [review]
Couple of comments and approved, also tested the UI and everything seems to work as it should.

**Pull request merged**